lib/cli/src/Cardano/CLI.hs

```diff
@@ -1014,6 +1016,16 @@ data WalletClient t = WalletClient
         -> ClientM NoContent
     , listPools
         :: ClientM [ApiStakePool]
+    , joinPool
+        :: ApiT PoolId
+        -> ApiT WalletId
+        -> ApiWalletData
+        -> ClientM NoContent
```

Comment:
Shouldn't the return type by a Transaction?

Reply:
we should definitely return something in `put` - I was thinking about 
```
data DelegateTransaction = DelegateTransaction {
  txId :: !(ApiT (Hash "Tx"))
, poolId :: !(ApiT PoolId)
, registeredSince :: !(Maybe ApiTimeReference)
}
```
what do you think?

Reply:
changed to `ApiTransaction` as there's no stake being delegated at this stage